src/presentation/table-presenter.ts

```diff
@@ -1,13 +1,15 @@
+import { QueryStatistics } from "../mesh/query-statistics.ts";
 import { MeshTable, TableGenerator } from "./mesh-table.ts";
 import { Presenter } from "./presenter.ts";
 
 export class TablePresenter implements Presenter {
   constructor(
     private readonly generator: TableGenerator,
     private readonly meshTable: MeshTable,
+    private readonly stats: QueryStatistics,
   ) {}
 
   present(): void {
-    this.meshTable.draw(this.generator);
+    this.meshTable.draw(this.generator, this.stats);
```

Comment:
d: thats what I meant, the change of variable between ctor to method. Actually having it in the ctor of the MeshTable avoids it to be put into this class ctor and reduces the argument count in this draw call. As a rule of thumb in OOP: usually a design choice that leads to reduction of passed arguments through classes means its a good choice because it has increased cohesion and reduced spaghetti code. By doing it via ctor all can be setup in the factories. The Presenter especially for the table are only a facade. You probably could even merge it with the `TableGenerator`